### DISCUSSION TITLE
Reading Capillary pressures from data file in PorousFlow

### DISCUSSION THREAD

**Discussion by sinaomrani96** [idaholab/moose#23413] [Q&A Modules: General] [answered]
Hi,
Hope you are fine.

I am new to MOOSE and its modules as well. I have read the documents and searched through discussions but did not find similar questions. I wanted to know if we can give our properties to moose? For example instead of using predefined Capillary Pressure functions i. e. constant, VG, BC, etc., we give a csv file containing CP vs saturation. Thanks.

Best regards,
Sina

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can read CSV files with a number of objects in MOOSE, but especially functions.
For example:
https://mooseframework.inl.gov/source/functions/PiecewiseLinear.html

Functions are usually dependent on time and position.
But in your code you can call functions with any arguments you like to replace time, x, y and z. 

Guillaume
